cctp_primitives/src/commitment_tree/mod.rs

```diff
@@ -128,22 +139,32 @@ impl CommitmentTree {
     // Adds Sidechain Creation Transaction to the Commitment Tree
     // Returns false if hash_scc can't get hash for data given in parameters;
     //         otherwise returns the same as set_scc_leaf method
-    pub fn add_scc(&mut self,
-                   sc_id: &[u8],
-                   amount: i64,
-                   pub_key: &[u8],
-                   withdrawal_epoch_length: u32,
-                   custom_data: &[u8],
-                   constant: Option<&[u8]>,
-                   cert_verification_key: &[u8],
-                   btr_verification_key: Option<&[u8]>,
-                   csw_verification_key: Option<&[u8]>,
-                   tx_hash: &[u8],
-                   out_idx: u32)-> bool {
+    pub fn add_scc(
+        &mut self,
+        sc_id: &[u8],
+        amount: u64,
+        pub_key: &[u8; 32],
+        tx_hash: &[u8; 32],
+        out_idx: u32,
+        withdrawal_epoch_length: u32,
+        cert_proving_system: u8,
+        csw_proving_system: u8,
+        mc_btr_request_data_length: u8,
+        custom_field_elements_configs: &[u8],
+        custom_bitvector_elements_configs: &[(u32, u32)],
+        btr_fee: u64,
+        ft_min_fee: u64,
+        // TODO: verify if it's enough to add to the comm_tree just the Poseidonhash of the custom_creation_data (Oleksandr)
+        custom_creation_data_hash: &[u8; FIELD_SIZE],
+        constant: Option<&[u8; FIELD_SIZE]>,
+        cert_verification_key_hash: &[u8; FIELD_SIZE],
+        csw_verification_key_hash: Option<&[u8; FIELD_SIZE]>
```

Comment:
For `csw_verification_key_hash` and `cert_verification_key_hash` why do we pass the hash? How to calculate the hash? Seems. better to expect for the origin key and calculate FE inside.

Reply:
We could reuse the fn pub hash_bytes() for computing the hash of a generic byte array (MC crypto lib is exporting such util func to cpp), but again I also think it would be better to pass origin bytes and get FE in rust.